src/hooks/useSession.ts

```diff
@@ -23,21 +34,27 @@ const useSession = (): DescopeSessionManager => {
       user: jwtResponse.user,
     }
     await DescopeReactNative.saveItem(projectId, JSON.stringify(updatedSession))
+    setCurrentTokens(updatedSession.sessionJwt, updatedSession.refreshJwt)
+    setCurrentUser(updatedSession.user)
     setSession(updatedSession)
   }
 
   const updateTokens = async (sessionJwt: string, refreshJwt: string) => {
     if (session) {
       logger?.log('updating current session with new tokens')
-      const updateSession = { ...session, sessionJwt, refreshJwt }
-      await DescopeReactNative.saveItem(projectId, JSON.stringify(updateSession))
-      setSession(updateSession)
+      const updatedSession = { ...session, sessionJwt, refreshJwt: refreshJwt || session.refreshJwt }
```

Comment:
is there a scenario that someone calls updateTokens with only session token? (e.i. empty refreshJwt)

Reply:
It's an edge case. In any way - the current session's refresh token is still viable